app_flutter/lib/logic/task_grid_filter.dart

```diff
@@ -0,0 +1,232 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/material.dart';
+
+import 'package:cocoon_service/protos.dart' show CommitStatus;
+
+import '../logic/qualified_task.dart';
+import '../widgets/filter_property_sheet.dart';
+
+/// A filter object for controlling which entries are visible in the Build dashboard grid
+/// of tasks. This filter object can filter on a number of properties of the tasks including
+/// the name of the task, information on the commit that generated the tasks, and platform
+/// stage type toggles.
+class TaskGridFilter extends FilterPropertySource {
+  TaskGridFilter();
```

Comment:
nit: Is there a use case where a Map is not passed to construct this?

Reply:
If you want a default/uncustomized filter.

I used to have a path that made use of that, but it isn't around any longer. I suppose you could use the map constructor with a null or empty map, but I figured something like this class should always support a default constructor as a general rule of thumb.

Reply:
This constructor is also used frequently in the tests.

Reply:
SGTM